### PR TITLE
add versioned golang.mk

### PR DESCRIPTION
**Pull request by natebrennand**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,39 +1,27 @@
...
-	@echo ""
-endif
-ifeq ($(COVERAGE),1)
-	@go test -cover -coverprofile=$(GOPATH)/src/$@/c.out $@ -test.v
-	@go tool cover -html=$(GOPATH)/src/$@/c.out
-endif
+$(PKGS): golang-test-all-deps
+	$(call golang-fmt,$@)
+	$(call golang-lint,$@)
+	$(call golang-vet,$@)
```
Comment (rgarcia) [inline Makefile]: why not `golang-test-all`?

**Comment by rgarcia** [comment]
lgtm besides one thing

File: Makefile (hunk 1 above)
Comment (natebrennand) [inline Makefile]: `integration_test.sh` wraps the `go test` invocation with starting up a dyamo server. It should be done w/ [this](https://github.com/Clever/images-private/tree/master/database/dynamodb), but that seems out of scope.
